### PR TITLE
CircleCI improvements

### PR DESCRIPTION
**Pull request by Warchant**
- [x] This PR adds new step to the CircleCI 2.0: `macos-build`.
       Mean time for build with cache: 2.30 min
       Mean time for build without cache: 11 min
- [x] Bugfixes in circle.yml related to sonar-pr and sonar-release steps.

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -112,21 +73,21 @@ jobs:
       - run:
           name: execute sonar-scanner to analyze PR
           command: >
-            [ ! -z "$SONAR_TOKEN" ] && \
+            if [ ! -z "$SONAR_TOKEN" ] && \
```
Comment (l4l) [inline .circleci/config.yml]: btw `! -z ...` is same as `-n`

File: .circleci/config.yml [hunk 2]
```diff
@@ -151,33 +112,77 @@ jobs:
...
-  #     - attach_workspace:
-  #         at: /opt/iroha/build
-  #     - run: TODO
-  #         name: todo
-  #         command: echo "I am happy albatross"
-
+  macos-build:
+    #docker:
+    #  - image: postgres:9.5
+    #  - image: redis:3.2.8
```
Comment (l4l) [inline .circleci/config.yml]: why commented?

File: .circleci/config.yml [hunk 3]
```diff
@@ -151,33 +112,77 @@ jobs:
...
+            - external-{{ arch }}-{{ .Branch }}
+            - external-{{ arch }}
+          paths:
+            - external
+      - run: ccache --show-stats
+      - run:
+          name: build
+          command: |
+            cmake -H. -Bbuild
+            cmake --build build -- -j$(sysctl -n hw.ncpu)
```
Comment (l4l) [inline .circleci/config.yml]: Afair we have some env variable for job number, don't we?

File: .circleci/config.yml [hunk 4]
```diff
@@ -151,33 +112,77 @@ jobs:
...
+          keys:
+            - external-{{ arch }}-{{ .Branch }}
+            - external-{{ arch }}
+          paths:
+            - external
+      - run: ccache --show-stats
+      - run:
+          name: build
+          command: |
+            cmake -H. -Bbuild
```
Comment (l4l) [inline .circleci/config.yml]: why not `cmake -H$IROHA_HOME -B$IROHA_BUILD`?

File: .circleci/config.yml (hunk 1 above)
Comment (Warchant) [reply .circleci/config.yml]: I've read somewhere that it is not.

File: .circleci/config.yml (hunk 2 above)
Comment (Warchant) [reply .circleci/config.yml]: Should I remove comments?

File: .circleci/config.yml (hunk 4 above)
Comment (Warchant) [reply .circleci/config.yml]: there are possibly no $IROHA_HOME or $IROHA_BUILD

File: .circleci/config.yml (hunk 3 above)
Comment (Warchant) [reply .circleci/config.yml]: in mac? are you sure?

File: .circleci/config.yml (hunk 2 above)
Comment (l4l) [reply .circleci/config.yml]: I mean why those containers isn't launched?

File: .circleci/config.yml (hunk 3 above)
Comment (l4l) [reply .circleci/config.yml]: In circleci, I guess it should share env vars to every arch, don't it?

Comment (Warchant) [reply .circleci/config.yml]: it does not

File: .circleci/config.yml (hunk 2 above)
Comment (Warchant) [reply .circleci/config.yml]: MacOS builds for circle ci 2.0 are bugged. When I uncomment it, circle ci thinks that I want to start linux, not mac... Everything fails.

**Pull request merged**